lib/Backend/DecompileFunction.cpp

```diff
@@ -450,29 +446,22 @@ std::string CCodeGenerator::buildAddressExpr(llvm::StringRef Expr) const {
          + addParentheses(Expr);
 }
 
-std::string
-CCodeGenerator::buildCastExpr(StringRef ExprToCast,
-                              const model::QualifiedType &SrcType,
-                              const model::QualifiedType &DestType) const {
-  if (SrcType == DestType or SrcType.UnqualifiedType().empty()
-      or DestType.UnqualifiedType().empty())
```

Comment:
I'm not sure what this piece of code originally meant.

`SrcType.UnqualifiedType().empty()` would mean that the whole type is unspecified?

@pfez

Reply:
A big chunk of the calls to this stems from the lookup on the `TypeMap` which can contain "unspecified" types for some cases.

Since I changed the arguments to use the "dereferenced" types - the need to check for that fell onto the callees instead (where we can mostly get away with an assert).

Reply:
Closing this thread.